server/lib/email.js

```diff
@@ -309,14 +311,31 @@ const generateEmailFromTemplate = (template, recipient, data = {}, options = {})
   return Promise.resolve(renderedTemplate);
 };
 
+const isNotificationActive = async (template, data) => {
+  // Skip active email notification check for unit test
+  if (config.env === 'test' || config.env === 'circleci') {
+    logger.info('Skips active email notification checks for unit test');
+    return true;
+  } else {
+    return models.Notification.isActive(template, data.collective.id, data.user.id);
```

Comment:
What happens if `data.collective` or `data.user` are null/undefined, can that happen?

Reply:
I don't expect that to happen but we probably need to be defensive with that, it happened with unit test but that was with a mocked data. I need suggestion on how to handle that, in case that happens, what do you think?

Reply:
@flickz There are notifications where `data.collective` is empty such as `user.monthlyreport`. This needs to be handled.

Let's be defensive on `data.user` and send if it's missing, without checking in the database.

Reply:
Alright noted.